### PR TITLE
Docs for docker builds with v1.1

### PR DESCRIPTION
**Pull request by jhermann**
Complete and ready for merge…

### CODE REVIEW

File: doc/examples/Dockerfile.build [hunk 1]
```diff
@@ -1,19 +1,91 @@
...
-        libssl-dev libncurses5-dev libncursesw5-dev libzmq3-dev \
-    && ( curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - ) \
-    && echo 'deb https://deb.nodesource.com/#NODEREPO# #CODENAME# main' \
+# Build arguments, as provided by 'build.sh'
+ARG DIST_ID="debian"
+ARG CODENAME="stretch"
+ARG PKGNAME
+
+# Other build arguments (adapt as needed)
+ARG NODEREPO="node_8.x"
```
Comment (nailor) [inline doc/examples/Dockerfile.build]: Whops, this is something that escaped me previously: where do we need the node repo?

Comment (jhermann) [reply doc/examples/Dockerfile.build]: See the how-to right above it. the sample project I took simply needs it.

and it is always easier to remove stuff than to add it.

Comment (nailor) [reply doc/examples/Dockerfile.build]: I'd be bit hesitant to add a custom debian repo by default in people's builds. I do agree that it is easier to remove things than to add, but I'm not comfortable pushing a custom repo by default, when it is actually not needed for the build process itself.

Comment (jhermann) [reply doc/examples/Dockerfile.build]: it is not obscure but nodesource, and these files are *tested* and working 1:1 copies out of a project.

Comment (nailor) [reply doc/examples/Dockerfile.build]: Yup, I understand that. I also don't think that particular custom debian repo is shady in any way. However this very documentation also ships with Debian (among others), and I'd like to keep it as lean as possible and only refer to sanctioned Debian repos.

It is really nice to have an example of the build file, but I'm wondering if we can find a simple example (which are, arguably, hard to fabricate) to fill the use case.

And down the line, people are going to just 1:1 copy these things and modify the few bits to fit their builds, so we could potentially introduce a pile of unnecessary steps in their build processes. Minor annoyance for sure, but still something

**Comment by jhermann** [comment]
I nonetheless need the *full* docs, thus the new master location is now here:

https://dockyard.readthedocs.io/en/latest/packaging-howto.html

You can either close this (and then still need to handle the existing 2 sections, e.g. link to the above), or merge and then edit as you see fit.

**Comment by nailor** [comment]
Hmm, yeah, maybe it makes more sense to link to the Dockyard docs as you said? That way we could avoid keeping two sets of docs up-to-date

**Comment by jhermann** [comment]
Replaced by #259

**Pull request closed without merging**